src/videojs-contrib-hls.js

```diff
@@ -139,13 +139,37 @@ export default class HlsHandler extends Component {
         this.options_[option] = this.source_[option];
       }
     });
-    this.masterPlaylistController_ = new MasterPlaylistController({
-      url: this.source_.src,
-      withCredentials: this.options_.withCredentials,
-      currentTimeFunc: this.tech_.currentTime.bind(this.tech_),
-      mediaSourceMode: this.options_.mode,
-      hlsHandler: this,
-      externHls: Hls
+    this.options_.url = this.source_.src;
+    this.options_.tech = this.tech_;
+    this.options_.externHls = Hls;
+    this.options_.bandwidth = this.bandwidth;
+    this.masterPlaylistController_ = new MasterPlaylistController(this.options_);
```

Comment:
I can understand that feeling. I was thinking the MPC is a "project private" sort of thing so it didn't need its own set of configuration options.